src/components/Page/PageRenderer.js

```diff
@@ -10,12 +10,36 @@ const renderStyles = (styles) => {
 const renderContent = (content) => React.isValidElement(content) && content.type === Page ? content : <Page>{content}</Page>;
 
 class PageRenderer extends Component {
+  constructor() {
+    super();
+    this.jump = this.jump.bind(this);
+  }
+
   componentDidMount() {
     this.context.catalog.page.scripts.forEach(runscript);
+    this.jump();
   }
 
   componentDidUpdate() {
     this.context.catalog.page.scripts.forEach(runscript);
+    this.jump();
+  }
+
+  jump() {
+    const {location: {query: {a}, hash}} = this.props;
```

Comment:
So does this only work for pushState then? I think that might be important to document.